qiskit_nature/algorithms/excited_states_solvers/qeom.py

```diff
@@ -95,14 +91,15 @@ def solve(self, driver: BaseDriver,
                            "evaluated on the ground state.")
 
         # 1. Run ground state calculation
-        groundstate_result = self._gsc.solve(driver, aux_operators)
+        groundstate_result = self._gsc.solve(problem)
 
         # 2. Prepare the excitation operators
-        matrix_operators_dict, size = self._prepare_matrix_operators()
+        matrix_operators_dict, size = self._prepare_matrix_operators(problem)
 
         # 3. Evaluate eom operators
         measurement_results = self._gsc.evaluate_operators(
-            groundstate_result.raw_result['eigenstate'],
+            groundstate_result.raw_result.eigenstate,
+            # TODO access depends on the interpret function
```

Comment:
We have the following code:

```
eigenstate_result = None
    if isinstance(raw_result, EigenstateResult):
        eigenstate_result = raw_result
    elif isinstance(raw_result, EigensolverResult):
        eigenstate_result = EigenstateResult()
        eigenstate_result.raw_result = raw_result
        eigenstate_result.eigenenergies = raw_result.eigenvalues
        eigenstate_result.eigenstates = raw_result.eigenstates
        eigenstate_result.aux_operator_eigenvalues = raw_result.aux_operator_eigenvalues
    elif isinstance(raw_result, MinimumEigensolverResult):
        eigenstate_result = EigenstateResult()
        eigenstate_result.raw_result = raw_result
        eigenstate_result.eigenenergies = np.asarray([raw_result.eigenvalue])
        eigenstate_result.eigenstates = [raw_result.eigenstate]
        eigenstate_result.aux_operator_eigenvalues = [raw_result.aux_operator_eigenvalues]
```
Sometimes there is `raw_result.eignestate` and sometimes `raw_result.eignestates`. The TODO is actually misleading but this is what I meant. Perhaps we should not refer to `raw_result` but to `eigenstate_result.eigenstates`. However it would require some more mods in the code.

Reply:
Aha I see. Because `eigenstate_result.eigenstates` wraps the actual eigen-state of the `raw_result`, regardless of whether it is named `eigenstate` or `eigenstates`.. How else would this affect the code?